openml-utils/src/main/java/com/feedzai/openml/util/jackson/deserializer/DatasetSchemaDeserializer.java

```diff
@@ -69,14 +70,17 @@ public DatasetSchema deserialize(final JsonParser jsonParser,
                                      final DeserializationContext deserializationContext) throws IOException {
```

Comment:
Have you confirmed if we can still deserialize a previous version in json to the new version with this logic?

Reply:
Since the fields of the DatasetSchema have not changed, I think the parameterized test covers both cases.

Reply:
From the parameterized test:
DatasetSchema with a target variable (same as the "old" version):

```json
{
  "targetIndex": 3,
  "fieldSchemas": [
    {
      "fieldIndex": 0,
      "fieldName": "field0",
      "valueSchema": {
        "@type": "numeric",
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 1,
      "fieldName": "field1",
      "valueSchema": {
        "@type": "numeric",
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 2,
      "fieldName": "field2",
      "valueSchema": {
        "@type": "numeric",
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 3,
      "fieldName": "field3",
      "valueSchema": {
        "@type": "categorical",
        "nominalValues": [
          "false",
          "true"
        ],
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 4,
      "fieldName": "field4",
      "valueSchema": {
        "@type": "categorical",
        "nominalValues": [
          "false",
          "true"
        ],
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 5,
      "fieldName": "field5",
      "valueSchema": {
        "@type": "string",
        "allowMissing": true
      }
    }
  ]
}
```

Without target variable:

```json
{
  "fieldSchemas": [
    {
      "fieldIndex": 0,
      "fieldName": "field0",
      "valueSchema": {
        "@type": "numeric",
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 1,
      "fieldName": "field1",
      "valueSchema": {
        "@type": "numeric",
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 2,
      "fieldName": "field2",
      "valueSchema": {
        "@type": "numeric",
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 3,
      "fieldName": "field3",
      "valueSchema": {
        "@type": "categorical",
        "nominalValues": [
          "false",
          "true"
        ],
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 4,
      "fieldName": "field4",
      "valueSchema": {
        "@type": "categorical",
        "nominalValues": [
          "false",
          "true"
        ],
        "allowMissing": true
      }
    },
    {
      "fieldIndex": 5,
      "fieldName": "field5",
      "valueSchema": {
        "@type": "string",
        "allowMissing": true
      }
    }
  ]
}
```

And to complete, one must just look at the travis green approval :P